go.mod

```diff
@@ -326,3 +326,5 @@ replace (
 	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.24.2
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.24.2
 )
+
+replace github.com/argoproj/gitops-engine => github.com/gdsoumya/gitops-engine v0.0.0-20230906104903-608e459b2a99
```

Comment:
Needs to be updated before merge, depends on https://github.com/argoproj/gitops-engine/pull/539